apps/address-service/domains/common/utils/services/suggest/providers/DadataSuggestionProvider.js

```diff
@@ -415,6 +420,71 @@ class DadataSuggestionProvider extends AbstractSuggestionProvider {
             }
         ))
     }
+
+    /**
+     * Checks token balance.
+     * In case of returned value less than DADATA_PROFILE_BALANCE_WARNING or any unhandled error -> returns false
+     * Otherwise returns true
+     * @return {Promise<boolean>}
+     */
+    async profileBalance () {
```

Comment:
Moving these new functions outside is better because they do not belong to suggestions and do not use encapsulated methods.
For example, you may add `utils/services/DadataHealtsChecker.js`. In this case, you even won't need to detect the provider.

Reply:
Done, check plz